pcdsdevices/aerotech.py

```diff
@@ -0,0 +1,693 @@
+            if print_move:
+                if wait:
+                    logger.info("Move completed for '%s'.", self.name)
+                else:
+                    logger.info("Move command sent to '%s'", self.name)
+            return status
+
+        # Catch all the common motor exceptions
+        except LimitError:
+            logger.warning("Requested move '%s' is outside the soft limits "
+                           "%s for motor %s", position, self.limits, self.name)
+        except MotorDisabledError:
+            logger.warning("Cannot move - motor %s is currently disabled. Try "
+                           "running 'motor.enable()'.", self.name)
+        except AerotechMotorFaulted:
+            logger.warning("Cannot move - motor %s is currently faulted. Try "
+                           "running 'motor.clear()'.", self.name)
+        except AerotechMotorStopped:
+            logger.warning("Cannot move - motor %s is currently stopped. Try "
+                           "running 'motor.state=\"Go\"'.", self.name)
+
+    def check_status(self, position=None):
+        """
+        Checks the status of the motor to make sure it is ready to move. Checks
+        the current position of the motor, and if a position is provided it
+        also checks that position.
+
+        Parameters
+        ----------
+        position : float, optional
+            Position to check for validity.
+
+        Raises
+        ------
+        MotorDisabledError
+            If the motor is disabled.
+
+        MotorFaulted
+            If the motor is faulted.
+
+        MotorStopped
+            If the motor is stopped.
+        """
+        if not self.enabled:
+            err = f"Motor '{self.name}' is currently disabled"
+            logger.error(err)
+            raise MotorDisabledError(err)
+
+        if self.faulted:
+            err = f"Motor '{self.name}' is currently faulted."
+            logger.error(err)
+            raise AerotechMotorFaulted(err)
+
+        if self.state == "Stop":
+            err = f"Motor '{self.name}' is currently stopped."
+            logger.error(err)
+            raise AerotechMotorStopped(err)
+
+        # Check if the current position is valid
+        self.check_value(self.position)
+        # Check if the move position is valid
+        if position:
+            self.check_value(position)
+
+    def set_position(self, position_des, print_set=True):
+        """
+        Sets the current position to be the inputted position by changing the
+        offset.
+
+        Parameters
+        ----------
+        position_des : float
+            The desired current position.
+        """
+        # Print to console or just to the log
+        if print_set:
+            log_level = logger.info
+        else:
+            log_level = logger.debug
+
+        log_level("'%s' previous position: %s, offset: %s", self.position,
+                  self.offset)
+        self.offset += position_des - self.position
```

Comment:
This does nothing in the current class hierarchy; I wonder if this relied on old pcdsdevices API?